src/webgpu/web-platform/copyImageBitmapToTexture.spec.ts

```diff
@@ -121,20 +121,33 @@ g.test('from ImageData')
     params()
       .combine(poptions('width', [1, 2, 4, 15, 255, 256]))
       .combine(poptions('height', [1, 2, 4, 15, 255, 256]))
+      .combine(poptions('alpha', ["none", "premultiply"]))
+      .combine(poptions('orientation', ["none", "flipY"]))
   )
   .fn(async t => {
-    const { width, height } = t.params;
+    const { width, height, alpha, orientation } = t.params;
 
     // The texture format is rgba8unorm, so the bytes per pixel is 4.
     const bytesPerPixel = 4;
 
     const imagePixels = new Uint8ClampedArray(bytesPerPixel * width * height);
-    for (let i = 0; i < width * height * bytesPerPixel; ++i) {
-      imagePixels[i] = i % 4 === 3 ? 255 : i % 256;
+    if (alpha === "premultiply") {
+      // Make expected value simple to construct
+      for (let i = 0; i < width * height * bytesPerPixel; ++i) {
+        imagePixels[i] = i % 4 != 3 ? 255 : i % 256;
```

Comment:
nit: parentheses would be more readable, prefer !== over !=
```suggestion
        imagePixels[i] = (i % 4 !== 3) ? 255 : i % 256;
```

Reply:
Linter may remove the parens if they're redundant. So do whatever the linter does for you (`grunt fix`), but prefer parens if it allows them.

Reply:
Comment in [1]

Reply:
Lint report error for the redundant parens. So I have to remove them, sry.